js_modules/dagster-ui/packages/ui-core/src/runs/RunTimeline.tsx

```diff
@@ -93,14 +93,18 @@ export const RunTimeline = (props: Props) => {
   const [_, end] = rangeMs;
   const includesTicks = now <= end;
 
-  const buckets = jobs.reduce(
-    (accum, job) => {
-      const {repoAddress} = job;
-      const repoKey = repoAddressAsURLString(repoAddress);
-      const jobsForRepo = accum[repoKey] || [];
-      return {...accum, [repoKey]: [...jobsForRepo, job]};
-    },
-    {} as Record<string, TimelineJob[]>,
+  const buckets = React.useMemo(
```

Comment:
This is really expensive so memoizing it

Reply:
does this work as expected ? when i was poking around here all the object references were shifting

Reply:
The object references change but at least this memo doesn't refire if the component re-renders for other reasons